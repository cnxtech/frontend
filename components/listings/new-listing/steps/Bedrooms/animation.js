import posed from 'react-pose'

const DELAY = 300

const getAnimatedComponent = (Component) =>
  posed(Component)({
    enter: { opacity: 1, delay: DELAY },
    exit: { opacity: 0 }
  })

export { getAnimatedComponent }
