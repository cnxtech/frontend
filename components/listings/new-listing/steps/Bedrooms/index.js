import React, { Component } from 'react'
import { Formik, Field } from 'formik'
import Button from '@emcasa/ui-dom/components/Button'
import Input from '@emcasa/ui-dom/components/Input'
import Row from '@emcasa/ui-dom/components/Row'
import Col from '@emcasa/ui-dom/components/Col'
import Container from 'components/listings/new-listing/shared/Container'
import Text from '@emcasa/ui-dom/components/Text'
import NavButtons from 'components/listings/new-listing/shared/NavButtons'

class Bedrooms extends Component {
  constructor(props) {
    super(props)
    this.nextStep = this.nextStep.bind(this)
    this.previousStep = this.previousStep.bind(this)
    this.validateBedroom = this.validateBedroom.bind(this)
    this.validateSuite = this.validateSuite.bind(this)
    this.validateBathroom = this.validateBathroom.bind(this)
    this.updateStateFromProps = this.updateStateFromProps.bind(this)
  }

  state = {
    bedrooms: 0,
    suites: 0,
    bathrooms: 0,
    enterMoreBedrooms: false,
    enterMoreBathrooms: false,
    showSuites: false,
    showBathrooms: false
  }

  componentDidMount() {
    this.updateStateFromProps(this.props)
  }

  componentWillReceiveProps(props) {
    this.updateStateFromProps(props)
  }

  updateStateFromProps(props) {
    const { rooms } = props
    if (rooms) {
      this.setState(rooms)
    }
  }

  nextStep() {
    const { navigateTo, updateRooms } = this.props
    updateRooms(this.state)
    navigateTo('garage')
  }

  previousStep() {
    const { navigateTo } = this.props
    navigateTo('homeDetails')
  }

  validateBedroom(value) {
    if (!value) {
      return 'É necessário informar o número de quartos.'
    }
    if (value < 0) {
      return 'Insira um valor positivo.'
    }
  }

  validateSuite(value) {
    if (typeof value !== 'number') {
      return 'É necessário informar o número de suítes.'
    }
    if (value < 0) {
      return 'Insira um valor positivo.'
    }
  }

  validateBathroom(value) {
    if (!value) {
      return 'É necessário informar o número de banheiros.'
    }
    if (value < 0) {
      return 'Insira um valor positivo.'
    }
  }

  bedroomSelection(setFieldTouched, setFieldValue, error, touched) {
    const { rooms } = this.props
    let bedrooms
    if (rooms) bedrooms = rooms.bedrooms
    if (this.state.enterMoreBedrooms) {
      return (
        <Col width={3/4}>
          <Input fluid hideLabelView placeholder="Número de quartos" type="number" error={touched ? error : null} onChange={(e) => {
            const { value } = e.target
            const intValue = parseInt(value)
            setFieldValue('bedroom', intValue)
            setFieldTouched('bedroom')
            this.setState({bedrooms: intValue, showSuites: true})
          }} defaultValue={bedrooms} />
        </Col>
      )
    }
    return (
      <Button.Group flexWrap="wrap" initialValue={bedrooms} onChange={(value) => {
          if (value === 'more') {
            this.setState({enterMoreBedrooms: true})
          } else {
            const intValue = parseInt(value)
            setFieldValue('bedroom', intValue)
            setFieldTouched('bedroom')
            this.setState({bedrooms: intValue, showSuites: true})
          }
        }}>
        <Button name="1" px={3} mr={2} value={1} height="tall">1</Button>
        <Button name="2" px={3} mr={2} value={2} height="tall">2</Button>
        <Button name="3" px={3} mr={2} value={3} height="tall">3</Button>
        <Button name="4" px={3} mr={2} value={4} height="tall">4</Button>
        <Button name="5" px={3} mr={2} value={5} height="tall">5</Button>
        <Button name="more" px={3} mr={2} value="more" height="tall">+</Button>
      </Button.Group>
    )
  }

  bathroomSelection(setFieldTouched, setFieldValue, error, touched) {
    const { rooms } = this.props
    let bathrooms
    if (rooms) bathrooms = rooms.bathrooms
    if (this.state.enterMoreBathrooms) {
      return (
        <Col width={3/4}>
          <Input fluid hideLabelView placeholder="Número de banheiros" type="number" error={touched ? error : null} onChange={(e) => {
            const { value } = e.target
            const intValue = parseInt(value)
            setFieldValue('bathroom', intValue)
            setFieldTouched('bathroom')
            this.setState({bathrooms: intValue})
          }} defaultValue={bathrooms} />
        </Col>
      )
    }
    return (
      <Button.Group flexWrap="wrap" initialValue={bathrooms} onChange={(value) => {
          if (value === 'more') {
            this.setState({enterMoreBathrooms: true})
          } else {
            const intValue = parseInt(value)
            setFieldValue('bathroom', intValue)
            setFieldTouched('bathroom')
            this.setState({bathrooms: intValue})
          }
        }}>
        <Button mr={2} px={2} value={1} height="tall">1</Button>
        <Button mr={2} px={3} value={2} height="tall">2</Button>
        <Button mr={2} px={3} value={3} height="tall">3</Button>
        <Button mr={2} px={3} value={4} height="tall">4</Button>
        <Button mr={2} px={3} value={5} height="tall">5</Button>
        <Button mr={2} px={3} value="more" height="tall">+</Button>
      </Button.Group>
    )
  }

  render() {
    const { rooms } = this.props
    let bedrooms, bathrooms, suites
    if (rooms) {
      bedrooms = rooms.bedrooms
      bathrooms = rooms.bathrooms
      suites = rooms.suites
    }
    return (
      <div ref={this.props.hostRef}>
        <Container>
          <Col width={[1,null,null,1/2]}>
            <Formik
              initialValues={{
                bedroom: bedrooms,
                suite: suites,
                bathroom: bathrooms
              }}
              isInitialValid={() => {
                return !(this.validateBedroom(bedrooms) && this.validateSuite(suites) && this.validateBathroom(bathrooms))
              }}
              render={({isValid, setFieldTouched, setFieldValue, errors}) => (
                <>
                  <Text
                    fontSize="large"
                    fontWeight="bold"
                    textAlign="center">
                    Quantos quartos?
                  </Text>
                  <Text color="grey">Quantos quartos tem no seu imóvel?</Text>
                  <Row mb={4} flexWrap="wrap">
                    <Field
                      name="bedroom"
                      validate={this.validateBedroom}
                      render={({form}) => this.bedroomSelection(setFieldTouched, setFieldValue, errors.bedroom, form.touched.bedroom)} />
                  </Row>
                  {this.state.showSuites && <> <Text color="grey">Algum deles é suíte? Quantos?</Text>
                    <Row mb={4} flexWrap="wrap">
                      <Field
                        name="suite"
                        validate={this.validateSuite}
                        render={() =>
                          <Button.Group flexWrap="wrap" initialValue={suites} onChange={(value) => {
                            setFieldValue('suite', value)
                            setFieldTouched('suite')
                            this.setState({suites: value, showBathrooms: true})
                            }}>
                            <Button mr={2} px={3} value={0} height="tall">Sem suíte</Button>
                            <Button mr={2} px={3} value={1} height="tall">1</Button>
                            <Button mr={2} px={3} value={2} height="tall">2</Button>
                            <Button mr={2} px={3} value={3} height="tall">3</Button>
                            <Button mr={2} px={3} value={4} height="tall">4</Button>
                          </Button.Group>
                        }/>
                    </Row>
                  </>}
                  {this.state.showBathrooms && <>
                    <Text color="grey">Sem contar lavabos e suítes, ele tem quantos banheiros?</Text>
                    <Row mb={4}>
                      <Field
                        name="bathroom"
                        validate={this.validateBathroom}
                        render={({form}) => this.bathroomSelection(setFieldTouched, setFieldValue, errors.bathroom, form.touched.bathroom)} />
                    </Row>
                  </>}
                  <NavButtons
                    previousStep={this.previousStep}
                    onSubmit={this.nextStep}
                    submitEnabled={isValid}
                  />
                </>
              )}
            />
          </Col>
        </Container>
      </div>
    )
  }
}

export default Bedrooms
