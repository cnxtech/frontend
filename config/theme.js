import theme from '@emcasa/ui'
import {fas} from '@fortawesome/pro-solid-svg-icons'
import {fal} from '@fortawesome/pro-light-svg-icons'
import icons from './icons'

export default {
  ...theme,
  icons: {emcasa: icons, default: fas, light: fal}
}
