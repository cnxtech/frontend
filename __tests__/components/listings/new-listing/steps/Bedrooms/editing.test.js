import React from 'react';
import Bedrooms from 'components/listings/new-listing/steps/Bedrooms';
import renderer from 'react-test-renderer';
import 'jest-styled-components'
import theme from '@emcasa/ui'
import { ThemeProvider } from 'styled-components'

describe('<AddressInput/>', () => {
  it('should render previously selected bedroom options', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <Bedrooms
          rooms={{
            bedrooms: 3,
            suites: 1,
            bathrooms: 3
          }}
        />
      </ThemeProvider>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
