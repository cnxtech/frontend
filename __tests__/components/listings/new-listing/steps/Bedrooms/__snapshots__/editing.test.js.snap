// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddressInput/> should render previously selected bedroom options 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 20px;
  padding-top: 0px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 20px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 20px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  outline: none;
  cursor: pointer;
  padding: 0 20px 0 20px;
  border-color: #e0e6ed;
  background-color: #ffffff;
  border-radius: 4px;
  border-width: 1px;
  width: auto;
  height: 60px;
  margin-right: 10px;
  padding-left: 14px;
  padding-right: 14px;
  color: #3c4858;
  font-family: FaktSoftPro-Normal;
  font-size: 18px;
  -webkit-font-smoothing: antialiased;
  border-style: solid;
}

.c6 :hover {
  background-color: #ffffff;
}

.c9 {
  outline: none;
  display: block;
  cursor: pointer;
  padding: 0 20px 0 20px;
  border-color: #e0e6ed;
  background-color: #ffffff;
  border-radius: 4px;
  border-width: 1px;
  width: 100%;
  height: 60px;
  color: #3c4858;
  font-family: FaktSoftPro-Normal;
  font-size: 18px;
  -webkit-font-smoothing: antialiased;
  border-style: solid;
}

.c9 :hover {
  background-color: #ffffff;
}

.c10 {
  outline: none;
  display: block;
  cursor: pointer;
  padding: 0 20px 0 20px;
  border-color: #f50057;
  background-color: #f50057;
  border-radius: 4px;
  border-width: 1px;
  width: 100%;
  height: 60px;
  color: #ffffff;
  font-family: FaktSoftPro-Normal;
  font-size: 18px;
  -webkit-font-smoothing: antialiased;
  border-style: solid;
}

.c10 :hover {
  background-color: #f50057;
}

.c2 {
  -webkit-font-smoothing: antialiased;
  font-family: FaktSoftPro-Normal;
  color: #3c4858;
  font-size: 24px;
  font-weight: bold;
  text-align: center;
}

.c3 {
  -webkit-font-smoothing: antialiased;
  font-family: FaktSoftPro-Normal;
  color: #979797;
  font-size: 18px;
}

.c1 {
  width: 100%;
}

.c8 {
  width: 41.66666666666667%;
}

@media screen and (min-width:780px) {
  .c1 {
    width: 50%;
  }
}

<div>
  <div
    className="c0"
  >
    <div
      className="c1"
      width={
        Array [
          1,
          0.5,
        ]
      }
    >
      <p
        className="c2"
        color="dark"
        fontSize="large"
        fontWeight="bold"
      >
        Quantos quartos?
      </p>
      <p
        className="c3"
        color="grey"
        fontSize="default"
      >
        Quantos quartos tem no seu imóvel?
      </p>
      <div
        className="c4"
      >
        <div
          className="c5"
          onChange={[Function]}
        >
          <button
            className="c6"
            fontSize="default"
            height="tall"
            name="1"
            onClick={[Function]}
            selected={false}
            value={1}
          >
            1
          </button>
          <button
            className="c6"
            fontSize="default"
            height="tall"
            name="2"
            onClick={[Function]}
            selected={false}
            value={2}
          >
            2
          </button>
          <button
            className="c6"
            fontSize="default"
            height="tall"
            name="3"
            onClick={[Function]}
            selected={true}
            value={3}
          >
            3
          </button>
          <button
            className="c6"
            fontSize="default"
            height="tall"
            name="4"
            onClick={[Function]}
            selected={false}
            value={4}
          >
            4
          </button>
          <button
            className="c6"
            fontSize="default"
            height="tall"
            name="5"
            onClick={[Function]}
            selected={false}
            value={5}
          >
            5
          </button>
          <button
            className="c6"
            fontSize="default"
            height="tall"
            name="more"
            onClick={[Function]}
            selected={false}
            value="more"
          >
            +
          </button>
        </div>
      </div>
      <div
        className="c7"
      >
        <div
          className="c8"
          width={0.4166666666666667}
        >
          <button
            className="c9"
            fontSize="default"
            height="tall"
            onClick={[Function]}
          >
            Voltar
          </button>
        </div>
        <div
          className="c8"
          width={0.4166666666666667}
        >
          <button
            className="c10"
            disabled={false}
            fontSize="default"
            height="tall"
            onClick={[Function]}
          >
            Avançar
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
